stile/sys_tests.py

```diff
@@ -902,6 +963,106 @@ class ScatterPlotSysTest(SysTest):
+        @param per_ccd_stat    Which statistics (median, mean, or None) to be performed within
+                               each CCD.
+                               [default: None, meaning no statistics is performed]
+        @param xlabel          The label of x-axis.
+                               [default: None, meaning do not show a label of x-axis]
+        @param ylabel          The label of y-axis.
+                               [default: None, meaning do not show a label of y-axis]
+        @param zlabel          The label of z values which appears at the side of color bar.
+                               [default: None, meaning do not show a label of z values]
+        @param color           The color of scattered points. This color is also applied to
+                               linear regression if argument `linear_regression` is True. This
+                               parameter is ignored when z is not None. In this case, the
+                               color of linear regression is set to blue.
+                               [default: None, meaning follow a matplotlib's default color]
+        @param lim             The limit of axis. This can be specified explicitly by
+                               using tuples such as ((xmin, xmax), (ymin, ymax)).
+                               If one passes float p, it calculate p%-percentile around median
+                               for each of x-axis and y-axis.
+                               [default: None, meaning do not set any limits]
+        @equal_axis            If True, force ticks of x-axis and y-axis equal to each other.
+                               [default: False]
+        @linear_regression     If True, perform linear regression for x and y and plot a
+                               regression line. If yerr is not None, perform the linear
+                               regression with incorporating the error into the standard chi^2
+                               and plot a regression line with a 1-sigma allowed region.
+                               [default: False]
+        @reference_line        Draw a reference line. If reference_line == 'one-to-one', x=y
+                               is drawn. If reference_line == 'zero', y=0 id drawn.
+                               A user-specific function can be used by passing an object which
+                               has an attribute '__call__' and returns a 1-d Numpy array.
+        @returns               a matplotlib.figure.Figure object
+        """
+        if per_ccd_stat:
+            if z_field is None:
+                z = None
+                x, y, yerr = self.getStatisticsPerCCD(array['CCD'], array[x_field],
+                                                      array[y_field], yerr = array[yerr_field],
+                                                      stat = per_ccd_stat)
+                self.data = stile_utils.FormatArray(numpy.rec.fromarrays([list(set(array['CCD'])), x,
```

Comment:
You don't need to run this through `FormatArray`, I think, since it's doing almost the same thing as `numpy.rec.fromarrays`--unless you tried it without and it didn't work?